workspaces/marketplace/plugins/marketplace/src/hooks/useExtensionReadConfigPermission.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useQuery } from '@tanstack/react-query';
+
+import { useMarketplaceApi } from './useMarketplaceApi';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { extensionPluginCreatePermission } from '@red-hat-developer-hub/backstage-plugin-marketplace-common';
+
+export const useExtensionReadConfigPermission = (
+  namespace: string,
+  name: string,
+) => {
+  const marketplaceApi = useMarketplaceApi();
+  return useQuery({
+    queryKey: ['marketplaceApi', 'getPluginConfigByName', namespace, name],
+    queryFn: () => marketplaceApi.getPluginConfigByName?.(namespace, name),
+  });
+};
+
+export const useExtensionInstallPluginPermission = () => {
+  const canInstallPlugin = usePermission({
+    permission: extensionPluginCreatePermission,
+    resourceRef: extensionPluginCreatePermission.resourceType,
+  });
```

Comment:
We recently found out that we have been using `usePermission` incorrectly. The `resourceRef` is suppose to point to the resource in question that you are doing the permission check against. See what I had to do for the [RBAC frontend plugin](https://github.com/backstage/community-plugins/blob/main/workspaces/rbac/plugins/rbac/src/components/RolesList/DeleteRole.tsx#L46) recently.